tool/builder/lib/sdks_finder.dart

```diff
@@ -0,0 +1,380 @@
+/// command helps by automating a portion of the task.
+///
+/// The cache-busting mechanism of Flutter's native assets feature is a hash
+/// of the contents of any build dependencies. The output files of this command
+/// are included in the build dependencies (as specified by the contents of each
+/// package's `build.dart` file), so if this command generates new SDK locations
+/// in those files, Flutter's CLI will have a cache miss, will re-run
+/// `build.dart` during the build phase, and in turn will download the newest
+/// versions of the MediaPipe SDKs onto the developer's machine.
+///
+/// Operationally, [SdksFinderCommand]'s implementation involves orchestrating
+/// one [_OsFinder] instance for each supported [OS] value, which in turn
+/// identifies the correct build that has been uploaded to GCS by MediaPipe
+/// release machinery.
+///
+/// Usage:
+/// ```sh
+///   $ cd path/to/flutter-mediapipe
+///   $ dart tool/builder/bin/main.dart sdks [-v]
+/// ```
+class SdksFinderCommand extends Command with RepoFinderMixin {
+  SdksFinderCommand() {
+    addVerboseOption(argParser);
+  }
+  @override
+  String description =
+      'Updates MediaPipe SDK manifest files for the current build target';
+
+  @override
+  String name = 'sdks';
+
+  static const _gcsPrefix = 'https://storage.googleapis.com';
+
+  /// Google Storage bucket which houses all MediaPipe SDK uploads.
+  static const _bucketName = 'mediapipe-nightly-public/prod/mediapipe';
+
+  final _finders = <_OsFinder>[
+    _OsFinder(OS.android),
+    _OsFinder(OS.macOS),
+    // TODO: Add other values as their support is ready
+  ];
+
+  @override
+  Future<void> run() async {
+    setUpLogging();
+    _checkGsUtil();
+    final results = _SdkLocations();
+
+    for (final finder in _finders) {
+      await for (final sdkLocation in finder.find()) {
+        results.add(sdkLocation);
+      }
+    }
+    for (final MediaPipeSdk sdk in MediaPipeSdk.values) {
+      _log.info('Saving locations for ${sdk.name}');
+      _writeResults(sdk, results.toMap(sdk));
+    }
+  }
+
+  void _writeResults(MediaPipeSdk sdk, _FlatResults results) {
+    final file = _getOutputFile(sdk);
+    _log.fine('Writing data to "${file.absolute.path}"');
+    var encoder = JsonEncoder.withIndent('  ');
+    file.writeAsStringSync('''// Generated file. Do not manually edit.
+final Map<String, Map<String, String>> sdkDownloadUrls = ${encoder.convert(results).replaceAll('"', "'")};
+''');
+  }
+
+  File _getOutputFile(MediaPipeSdk sdk) {
+    return File(
+      path.joinAll([
+        findFlutterMediaPipeRoot().absolute.path,
+        'packages/${sdk.toPackageDir()}',
+        'sdk_downloads.dart',
+      ]),
+    );
+  }
+
+  void _checkGsUtil() async {
+    if (!io.Platform.isMacOS && !io.Platform.isLinux) {
+      // `which` is not available on Windows, so allow the command to attempt
+      // to run on Windows
+      // TODO: possibly add Windows-specific support
+      return;
```

Comment:
What will happen in `run` after this method is called if this method no-ops on Windows? Should we just throw?

Reply:
If `gsutil` is installed and on their path, the command will succeed. If `gsutil` is missing, they will get whatever error Windows emits when you attempt to run a command that it knows nothing about. All they will miss is the targeted "Hey, go to this URL to install this thing you need" message.